src/Compilers/CSharp/Portable/Errors/ErrorCode.cs

```diff
@@ -1609,6 +1609,8 @@ internal enum ErrorCode
         ERR_NotNullableOperatorNotReferenceType = 8624,
         WRN_NullAsNonNullable = 8625,
         WRN_NoBestNullabilityConditionalExpression = 8626,
+        ERR_AttributeRequiresBoolReturn = 8627,
```

Comment:
>ERR_AttributeRequiresBoolReturn [](start = 8, length = 31)

I didn't notice where these are used, but I'm not confident it is a good idea to have diagnostics such as this.

Reply:
Because of "forward compatibility" scenarios. (Code that compiles with older compilers will fail to compile with newer compilers)

---
In reply to: [187484195](https://github.com/dotnet/roslyn/pull/26656#discussion_r187484195) [](ancestors = 187484195)

Reply:
Thanks for the clarification.
Filed https://github.com/dotnet/roslyn/issues/26783 to track compat concern.